src/hmac/index.ts

```diff
@@ -1,7 +1,7 @@
 import crypto from 'crypto'
 import lengths from './lengths.js'
 
-export async function create (hash: 'SHA1' | 'SHA256' | 'SHA512', secret: Uint8Array) { // eslint-disable-line require-await
+export async function create (hash: 'SHA1' | 'SHA256' | 'SHA512', secret: Uint8Array): Promise<{ digest: (data: Uint8Array) => Promise<Buffer>, length: number }> {
```

Comment:
When you have a type like this with multiple fields it's usually better to export an interface and use that in the return type:

```suggestion
export interface Hmac {
  digest: (data: Uint8Array) => Promise<Uint8Array>
  length: number
}

export async function create (hash: 'SHA1' | 'SHA256' | 'SHA512', secret: Uint8Array): Promise<Hmac> {
```